packages/remote-feature-flag-controller/src/remote-feature-flag-controller.test.ts

```diff
@@ -0,0 +1,241 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+
+import type { AbstractClientConfigApiService } from './client-config-api-service/abstract-client-config-api-service';
+import {
+  RemoteFeatureFlagController,
+  controllerName,
+} from './remote-feature-flag-controller';
+import type {
+  RemoteFeatureFlagControllerActions,
+  RemoteFeatureFlagControllerMessenger,
+  RemoteFeatureFlagControllerState,
+  RemoteFeatureFlagControllerStateChangeEvent,
+} from './remote-feature-flag-controller';
+
+const mockFlags = {
+  feature1: true,
+  feature2: { chrome: '<109' },
+};
+
+/**
+ * Creates a controller instance with default parameters for testing
+ */
+/**
+ * Creates a controller instance with default parameters for testing
+ * @param options - The controller configuration options
+ * @param options.messenger - The controller messenger instance
+ * @param options.state - The initial controller state
+ * @param options.clientConfigApiService - The client config API service instance
+ * @param options.disabled - Whether the controller should start disabled
+ * @returns A configured RemoteFeatureFlagController instance
+ */
+function createController(
+  options: Partial<{
+    messenger: RemoteFeatureFlagControllerMessenger;
+    state: Partial<RemoteFeatureFlagControllerState>;
+    clientConfigApiService: AbstractClientConfigApiService;
+    disabled: boolean;
+  }> = {},
+) {
+  return new RemoteFeatureFlagController({
+    messenger: options.messenger ?? getControllerMessenger(),
+    state: options.state ?? {},
+    clientConfigApiService:
+      options.clientConfigApiService ?? buildClientConfigApiService(),
+    disabled: options.disabled,
+  });
+}
+
+describe('RemoteFeatureFlagController', () => {
+  let clientConfigApiService: AbstractClientConfigApiService;
+
+  beforeEach(() => {
+    clientConfigApiService = buildClientConfigApiService();
+    jest.clearAllMocks();
+  });
+
+  describe('constructor', () => {
+    it('should initialize with default state', () => {
+      const controller = createController();
+
+      expect(controller.state).toStrictEqual({
+        remoteFeatureFlags: {},
+        cacheTimestamp: 0,
+      });
+    });
+
+    it('should initialize with disabled parameter', () => {
+      const controller = createController({ disabled: true });
+
+      expect(controller.state).toStrictEqual({
+        remoteFeatureFlags: {},
+        cacheTimestamp: 0,
+      });
+    });
+  });
+
+  describe('getRemoteFeatureFlags', () => {
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should get feature flags from API when cache is invalid', async () => {
+      const controller = createController({ clientConfigApiService });
```

Comment:
What in this test is making the cache invalid?

If it's the default data that we are having the service object return, what are your thoughts on including that data in this test? Otherwise it's not totally clear. Something like:

``` suggestion
      const controller = createController({ clientConfigApiService });

      jest
        .spyOn(clientConfigApiService, 'fetchFlags')
        .mockImplementation(async () => ({
          error: false,
          message: 'Success',
          statusCode: '200',
          statusText: 'OK',
          cachedData: { differentFlag: true },
          cacheTimestamp: Date.now(),
        }));
```

Reply:
There are some duplicated test cases. I had a clean up and capture cases with cache expired in these two, hope it makes sense 🙏🏻 66ebdbeb6f0108b46c47df2338123cb4d087799b
```
- should make network request to fetch when cache is not expired
 this will return previous response even 2nd fetch is meant to be a different value
- should make network request to fetch when cache is expired
an expired timestamp is given and return the new value
```